Parent/Parent/Details/AssignmentDetailsViewController.swift

```diff
@@ -62,8 +78,7 @@ class AssignmentDetailsViewController: AssignmentDetailViewController {
 
         prepare(observer, refresher: refresher, detailsFactory: EventDetailsViewModel.detailsForAssignment(session.baseURL, observeeID: studentID, context: self))
 
-        disposable = observer.signal.map { $0.1 }
-            .observeValues { _ in
+        disposable = observer.signal.map { $0.1 }.observeValues { _ in
```

Comment:
this worked just fine, when it fired, not sure why this would only fire sometimes ????

Reply:
It will only fire when the assignment changes. You might have to also check `observer.object` to get the cached version.